pandas/conftest.py

```diff
@@ -453,3 +453,37 @@ def mock():
         return importlib.import_module("unittest.mock")
     else:
         return pytest.importorskip("mock")
+
+
+# ----------------------------------------------------------------
+# Global setup for tests using Hypothesis
+
+from hypothesis import strategies as st
```

Comment:
My read of the reviews so far is that @jreback was in favor of a mandatory dependency (also my recommendation), and you're now in favor too.

I've therefore made the relevant changes and it's all ready to go :tada: 

(though one build on Travis has errored out, the tests passed until the timeout)

Reply:
so, I still think we need to a) remove hypothesis from 1 build (the same one we have removed ``moto`` from is good). and use ``pyimportor.skip('hypthoesis')``. The reason is not for our CI really, rather so when a user does ``pd.test()`` is doesn't fail, rather it will just skip those tests.

Reply:
@jreback there are two problems with making Hypothesis optional for `pd.test()`:

1. It makes adding further Hypothesis tests - eg for serialisation round-trips, timedeltas, or reshaping logic - much harder.  They'd have to be in separate files, guard any global setup and configuration, handle import-or-skips, etc.
2. It forces us to choose to either duplicate tests, or skip them at runtime.

That doesn't make it completely unreasonable, I'd prefer to just have the dependency - and I've been using Pandas for much longer than Hypothesis!

TLDR - what's wrong with putting Hypothesis in the same category as pytest?

Reply:
I think moto is a bit different, since it's relatively unimportant to mainline pandas, and so is easy to work around.

IMO, hypothesis should be treated the same as pytest.